colorfield/fields.py

```diff
@@ -50,15 +52,28 @@ def __init__(self, *args, **kwargs):
         else:
             kwargs.setdefault('default', DEFAULT_PER_FORMAT[self.format])
         super(ColorField, self).__init__(*args, **kwargs)
+        
+        if self.choices and self.samples:
+            raise ImproperlyConfigured("You can only set one of 'choices' and 'samples' for ColorFields.")
 
     def formfield(self, **kwargs):
         palette = []
         if self.choices:
             choices = self.get_choices(include_blank=False)
             palette = [choice[0] for choice in choices]
+        elif self.samples:
+            palette = [choice[0] for choice in self.samples]
         kwargs['widget'] = ColorWidget(attrs={
             'default': self.get_default(),
             'format': self.format,
             'palette': palette,
+            'disable_picker': bool(self.choices),
```

Comment:
This is not used, is it for hiding the spectrum right?

Reply:
Ah, sorry. I left this by accident while I was playing around with hiding the picker, but nothing came of it

Reply:
Not sure how to disable the picker to  be honest - I played around with it a bit, the only thing I though up was to maybe feed an extra context variable to the html template file, embed it somewhere in the color field and act on it in `colorfield.js` (hide spectrum, adjust height of color picker element to fit only the palette) - wasn't successful so far.. in my free time I will play around a bit more and see if I can figure this out